From 5675a53c611fe4991d9f91d40fe6bd85ae74b5d6 Mon Sep 17 00:00:00 2001
From: Srikanth Thokala <srikanth.thokala@xilinx.com>
Date: Wed, 18 Jun 2014 16:04:54 +0530
Subject: [PATCH 189/793] vdma: xilinx: Use dma_slave_config generic API

This patch creates a separate function for the IP specific
configuration instead of embedding the driver specific configuration
into generic structure 'dma_slave_config'.

This patch also forces a change in dma test client.

Signed-off-by: Srikanth Thokala <sthokal@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/dma/xilinx/vdmatest.c |    6 ++----
 1 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/dma/xilinx/vdmatest.c b/drivers/dma/xilinx/vdmatest.c
index 83e8153..6210bd5 100644
--- a/drivers/dma/xilinx/vdmatest.c
+++ b/drivers/dma/xilinx/vdmatest.c
@@ -320,12 +320,10 @@ static int xilinx_vdmatest_slave_func(void *data)
 		config.frm_cnt_en = 1;
 		config.coalesc = frm_cnt * 10;
 		config.park = 1;
-		tx_dev->device_control(tx_chan, DMA_SLAVE_CONFIG,
-					(unsigned long)&config);
+		xilinx_vdma_channel_set_config(tx_chan, &config);
 
 		config.park = 0;
-		rx_dev->device_control(rx_chan, DMA_SLAVE_CONFIG,
-					(unsigned long)&config);
+		xilinx_vdma_channel_set_config(rx_chan, &config);
 
 		rxd = rx_dev->device_prep_slave_sg(rx_chan, rx_sg, frm_cnt,
 				DMA_DEV_TO_MEM, flags, NULL);
-- 
1.7.5.4

